src/planwise/component/projects2.clj

```diff
@@ -32,10 +34,17 @@
                                       :state "draft"}))
 
 (defn update-project
-  [store project]
-  (db-update-project (get-db store) (-> project
-                                        (update :config pr-str)
-                                        (dissoc :state))))
+  [store {:keys [config provider-set-id] :as project}]
+  (let [algorithm (-> (providers-set/get-provider-set (:providers-set store) provider-set-id)
+                      :coverage-algorithm
+                      keyword)
+        valid-criteria? (s/valid? ::coverage/criteria
+                                  (assoc (get-in project [:config :coverage :filter-options])
+                                         :algorithm algorithm))
```

Comment:
Can you add this validation check to the `start-project` function? That way, even if something's wrong in the client side, we still don't start the project with an invalid coverage configuration. Currently it's making a spec check, but since the coverage algorithm is in the provider set and not in the project, it can't verify that the coverage options are actually correct.

Reply:
Shouldn't specs also validate that criteria has only one filter option?  
Maybe using  :max-count 1 https://github.com/clojure/spec.alpha/blob/b2b5db433cad9f1ad56bd28f58 4136cef2bae73e/src/main/clojure/clojure/spec/alpha.clj#L545

Reply:
No, not really, because coverage algorithms may require more than one parameter.